packages/solana/lib/src/rpc/dto/parsed_message/header.dart

```diff
@@ -0,0 +1,28 @@
+/// The header of a [ParsedMessage]
+class Header {
+  Header({
+    required this.numRequiredSignatures,
+    required this.numReadonlySignedAccounts,
+    required this.numReadonlyUnsignedAccounts,
+  });
+
+  factory Header.fromJson(dynamic json) {
+    if (json is List) {
+      return Header(
+        numRequiredSignatures: json[0] as int,
+        numReadonlyUnsignedAccounts: json[1] as int,
+        numReadonlySignedAccounts: json[2] as int,
+      );
+    } else {
+      return Header(
```

Comment:
You can combine manual and generated code:

```dart
@JsonSerializable(createToJson: false)
class Header {
  Header({
    required this.numRequiredSignatures,
    required this.numReadonlySignedAccounts,
    required this.numReadonlyUnsignedAccounts,
  });

  factory Header.fromJson(dynamic json) {
    if (json is List) {
      return Header(
        numRequiredSignatures: json[0] as int,
        numReadonlyUnsignedAccounts: json[1] as int,
        numReadonlySignedAccounts: json[2] as int,
      );
    } else {
      return _$HeaderFromJson(json as Map<String, dynamic>);
    }
  }

  final int numRequiredSignatures;
  final int numReadonlySignedAccounts;
  final int numReadonlyUnsignedAccounts;
}
```

Reply:
I love it ;)